core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -87,15 +91,20 @@ SdkPojo unmarshall(XmlUnmarshallerContext context, SdkPojo sdkPojo, XmlElement r
 
                 if (isAttribute(field)) {
                     root.getOptionalAttributeByName(field.unmarshallLocationName())
-                        .ifPresent(e -> field.set(sdkPojo, e));
+                            .ifPresent(e -> field.set(sdkPojo, e));
                 } else {
                     List<XmlElement> element = isExplicitPayloadMember(field) ?
-                                               singletonList(root) :
-                                               root.getElementsByName(field.unmarshallLocationName());
+                            singletonList(root) :
+                            root.getElementsByName(field.unmarshallLocationName());
 
                     if (!CollectionUtils.isNullOrEmpty(element)) {
-                        Object unmarshalled = unmarshaller.unmarshall(context, element, (SdkField<Object>) field);
-                        field.set(sdkPojo, unmarshalled);
+                        if (payloadMemberAsBlobType.isPresent() && payloadMemberAsBlobType.get().equals(field)
```

Comment:
This is similar to [JsonProtocolUnmarshaller](https://github.com/aws/aws-sdk-java-v2/blob/35267ca707c3fb5cdf7e3e98758a8ef969269183/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java#L207-L209)